pkg/middleware/loggermw/logger.go

```diff
@@ -154,8 +154,36 @@ func errorLogParams(err error) []any {
 	}
 }
 
-var sensitiveQueryStrings = [...]string{
-	"auth_token",
+type queryCheck struct {
+	key     string
+	checker func(v url.Values) bool
+}
+
+var sensitiveQueryChecks = [...]queryCheck{
+	{
+		key: "auth_token",
+		checker: func(v url.Values) bool {
+			return v.Has("auth_token")
+		},
+	},
+	{
+		key: "X-Amz-Signature",
+		checker: func(v url.Values) bool {
+			return v.Has("X-Amz-Signature")
+		},
+	},
+	{
+		key: "X-Goog-Signature",
+		checker: func(v url.Values) bool {
+			return v.Has("X-Goog-Signature")
+		},
+	},
+	{
+		key: "sig",
+		checker: func(v url.Values) bool {
+			return v.Has("sig") && v.Has("sv")
+		},
+	},
 }
 
 func SanitizeURL(s string) (string, error) {
```

Comment:
Seems like we could update this to be more specific.

```suggestion
func SanitizeURI(s string) (string, error) {
```